crates/sui-types/src/temporary_store.rs

```diff
@@ -1203,31 +1265,62 @@ impl<S: GetModule + ObjectStore + BackingPackageStore> TemporaryStore<S> {
             .map(|(_, summary)| summary.clone())
             .unwrap_or_default();
 
-        // storage gas cost should be equal to total rebates of mutated objects + storage fund rebate inflow (see below).
-        // note: each mutated object O of size N bytes is assessed a storage cost of N * storage_price bytes, but also
-        // has O storage_rebate credited to the tx storage rebate.
-        // If we run out of gas we do not perform any storage charges and the assert will fail
-        // `gas_summary.storage_cost` is a good proxy for OOG
-        if gas_summary.storage_cost > 0 {
-            assert_eq!(gas_summary.storage_cost, output_rebate_amount);
+        if do_expensive_checks {
+            // note: storage_cost flows into the storage_rebate field of the output objects, which is why it is not accounted for here.
+            // similarly, all of the storage_rebate *except* the storage_fund_rebate_inflow gets credited to the gas coin
+            // both computation costs and storage rebate inflow are
+            total_output_sui +=
+                gas_summary.computation_cost + gas_summary.non_refundable_storage_fee;
+            if let Some((epoch_fees, epoch_rebates)) = advance_epoch_gas_summary {
+                total_input_sui += epoch_fees;
+                total_output_sui += epoch_rebates;
+            }
+            if total_input_sui != total_output_sui {
+                return Err(ExecutionError::invariant_violation(
+                format!("SUI conservation failed: input={}, output={}, this transaction either mints or burns SUI",
+                total_input_sui,
+                total_output_sui))
+            );
+            }
         }
 
-        // note: storage_cost flows into the storage_rebate field of the output objects, which is why it is not accounted for here.
-        // similarly, all of the storage_rebate *except* the storage_fund_rebate_inflow gets credited to the gas coin
-        // both computation costs and storage rebate inflow are
-        total_output_sui += gas_summary.computation_cost + gas_summary.non_refundable_storage_fee;
-        if let Some((epoch_fees, epoch_rebates)) = advance_epoch_gas_summary {
-            total_input_sui += epoch_fees;
-            total_output_sui += epoch_rebates;
-        };
-        fp_ensure!(
-            total_input_sui == total_output_sui,
-            SuiError::from(format!(
-                "SUI conservation violated: input={}, output={}, this transaction either mints or burns SUI",
-                total_input_sui,
-                total_output_sui,
-            ).as_str())
-        );
+        // If we run out of gas we do not perform any storage charges and the checks below will fail
```

Comment:
@dariorussi not sure about this part after our conversation on Friday, but tests definitely fail if I get rid of this and let the checks run

Reply:
ok getting this PR locally and check